tensorly/regression/cp_plsr.py

```diff
@@ -256,11 +301,11 @@ def transform(self, X, Y=None):
 
                 Y -= T.dot(
                     T.dot(
-                        T.lstsq(T.transpose(X_scores), T.transpose(X_scores))[0],
-                        Y_scores[:, [component]],
+                        X_scores,
+                        T.reshape(self.coef_[:, component], (-1, 1)),
```

Comment:
Do we need the reshape here?

Reply:
`T.dot` will somehow report error on shape if it is not in a matrix.